src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/JsonObject.cs

```diff
@@ -791,6 +856,32 @@ private static object ProcessCustomObject<T>(object o, int depth, in ConvertToJs
             return result;
         }
 
+        /// <summary>
+        /// When IgnoreNullProperties is set removes all null value properties from JObject.
+        /// Otherwise the original object with all properties is returned.
+        /// </summary>
+        private static object RemoveNullProperties(JObject jobject, in ConvertToJsonContext context)
+        {
+            if (context.IgnoreNullProperties)
+            {
+                List<string> nullProperties = new List<string>();
+                foreach (KeyValuePair<string, JToken> entry in jobject)
+                {
+                    if (entry.Value.Type == JTokenType.Null)
+                    {
+                        nullProperties.Add(entry.Key);
+                    }
+                }
+
+                foreach (string keyToRemove in nullProperties)
+                {
+                    jobject.Remove(keyToRemove);
+                }
+            }
+
+            return jobject;
```

Comment:
I am not sure that we want to change source object.

Reply:
I think I misunderstood what you meant before.

> Сan we not create a new object but delete it from the original?

Should I deep clone the object before?

Reply:
BTW, you are right, this is changing the source object.

Reply:
Sorry, if my previous comment confused you.

Reply:
No problem :)